File: surveyplanet-api-test.py
import requests
import base64
import sys
import json

API_HOST = 'api.surveyplanet.com'

CLIENT_ID = sys.argv[1]
CLIENT_SECRET = sys.argv[2]

if not CLIENT_ID or not CLIENT_SECRET:
    print('Usage: python script.py <client_id> <client_secret>')
    sys.exit(1)

def get_auth_token():
    AUTH = base64.b64encode(f"{CLIENT_ID}:{CLIENT_SECRET}".encode()).decode()

    headers = {
        'Content-Type': 'application/x-www-form-urlencoded',
        'Accept': 'application/json',
        'Authorization': f'Basic {AUTH}',
    }
    data = {
        'grant_type': 'client_credentials'
    }

    response = requests.post(f'https://{API_HOST}/oauth/token', headers=headers, data=data)

    if response.status_code == 200:
        return response.json()['data']['access_token']
    else:
        print(f"Status code: {response.status_code}")
        print(f"Response text: {response.text}")
        raise Exception('Cannot get auth token')

def get_user_data(token):
    headers = {
        'Accept': 'application/json',
        'Authorization': f'Bearer {token}',
    }

    response = requests.get(f'https://{API_HOST}/v1/user', headers=headers)

    if response.status_code == 200:
        return response.json()
    else:

        print(f"Status code: {response.status_code}")
        print(f"Response text: {response.text}")
        raise Exception('Cannot get user data')

try:
    token = get_auth_token()
    user_data = get_user_data(token)
    print(json.dumps(user_data, indent=4))
except Exception as e:
    print(e)

File: surveyplanet-api-test.mjs
import https from 'https';
import querystring from 'querystring';

const API_HOST = 'api.surveyplanet.com';

const CLIENT_ID = process.argv[2];
const CLIENT_SECRET = process.argv[3];

if (!CLIENT_ID || !CLIENT_SECRET) {
	console.log('Usage: node script.js <client_id> <client_secret>');
	process.exit(1);
}

const getAuthToken = async () => {

	const AUTH = Buffer.from(`${CLIENT_ID}:${CLIENT_SECRET}`)
		.toString( 'base64' );

	const options = {
		hostname: API_HOST,
		path: '/oauth/token',
		method: 'POST',
		headers: {
			'Content-Type': 'application/x-www-form-urlencoded',
			Accept: 'application/json',
			Authorization: `Basic ${AUTH}`,
		},
	};

	return new Promise((resolve, reject) => {
		const req = https.request(options, (res) => {
			let data = '';

			res.on('data', (chunk) => {
				data += chunk;
			});

			res.on('end', () => {
				const AUTH_RES_JSON = JSON.parse(data);

				if (
					!AUTH_RES_JSON ||
					!AUTH_RES_JSON.data ||
					!AUTH_RES_JSON.data.access_token ||
					!AUTH_RES_JSON.data.access_token.length
				) {
					return reject(new Error('Cannot get auth token'));
				}

				return resolve(AUTH_RES_JSON.data.access_token);
			});
		});

		req.on('error', (error) => {
			return reject(error);
		});

		req.write(querystring.stringify({ grant_type: 'client_credentials' }));
		req.end();
	});
}

const getUserData = async (token) => {

	const options = {
		hostname: API_HOST,
		path: '/v1/user',
		method: 'GET',
		headers: {
			Accept: 'application/json',
			Authorization: `Bearer ${token}`,
		},
	};

	return new Promise((resolve, reject) => {
		const req = https.request(options, (res) => {
			let data = '';

			res.on('data', (chunk) => {
				data += chunk;
			});

			res.on('end', () => {
				return resolve(JSON.parse(data));
			});
		});

		req.on('error', (error) => {
			return reject(error);
		});

		req.end();
	});
}


( async () => {
	const token = await getAuthToken();
	const userData = await getUserData(token);
	return userData;
})()
.then( (result) => {
	console.log(result);
	process.exit(0);
})
.catch( (e) => {
	console.error(e);
	process.exit(1);
})
